content/blog/open-source-monitoring-at-the-hpe-customer-technology-center-böblingen.md

```diff
@@ -0,0 +1,377 @@
+      <server> <ILO_ip>10.0.44.13</ILO_ip>,<Rack>F12</Rack>,<Loc>20</Loc> </server>     
+      <server> <ILO_ip>10.1.24.66</ILO_ip>,<Rack>C5</Rack>,<Loc>30</Loc> </server>     
+      <server> <ILO_ip>10.0.43.14</ILO_ip>,<Rack>i12</Rack>,<Loc>9</Loc> </server>
+    </data>
+```
+
+As you can see the config map contains two data parts: the iloprometheus.key and the iloprometheus.xml section. The first one is storing the encryption key, while the second one is storing user and password encrypted with the encryption key iloprometheus.key, the monitoring intervall, the logfile name and the port used by the exporter together with the server data. I used the rack location in addition to the ILO IP address in order to be able to visualize a rack temperature diagram if needed.
+
+T﻿he config map was stored in a YAML-file (iloprom1.yml) and then applied in my Kubernetes environment:
+
+```shell
+kubectl apply -f iloprom1.yml
+```
+
+With the config map in place, I could define the Kubernetes pod for the HPE Server exporter. The pod is using the iloprometheus:v4.3.1 container image, mounting the config map data to the /opt/prometheus/data directory in the container and is starting the Python script /opt/prometheus/iloPromConnector.v4.3.py.
+
+```shell
+cat << 'EOF' | kubectl apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ilo1
+  namespace: svtprometheus
+  
+  labels:
+    app: ilo1
+spec:
+  selector:
+    matchLabels:
+      app: ilo1
+  template:
+    metadata:
+      labels:
+        app: ilo1
+    spec:
+      containers:
+        - name: ilo1
+          image: tb1378/iloprometheus:v4.3.1
+          command: ["/usr/bin/python3"]
+          args: ["/opt/prometheus/iloPromConnector.v4.3.py"]
+          volumeMounts:
+            - name: iloprometheusxml
+              mountPath: /opt/prometheus/data
+      volumes:
+        - name: iloprometheusxml
+          configMap:
+            name: iloprom1-xml   # the correct name of the configmap needs to be added here. 
+EOF
+```
+
+After checking that the pod is up and running, I still needed to define a Kubernetes service using this pod/app in order to have at the end the exporter data available for Prometheus.
+
+```shell
+cat << 'EOF' | kubectl apply -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: ilo1
+  namespace: svtprometheus
+#  labels: 
+#     hpecp.hpe.com/hpecp-internal-gateway: "true"    
+spec:
+  selector:
+    app: ilo1
+  ports:
+    - port: 9091               # The Port of that the Prometheus connector uses
+      targetPort: 9091
+      protocol: TCP
+  type: NodePort               # expose the Prometheus connector if you want/need to debug it. 
+EOF
+```
+
+Next, I updated the Prometheus config map to make sure, that the server exporter data is collected and stored in the Prometheus time series database. Hence, I added a job for the server exporter to the existing job list of the Prometheus config map:
```

Comment:
Remove the comma before "that the server exporter".
Change "is collected" to "was collected"